src/components/TeacherComponents/hooks/downloadAndShareHooks/useResourceFormState.tsx

```diff
@@ -97,10 +91,57 @@ export const useResourceFormState = (props: UseResourceFormStateProps) => {
   } = schoolFromLocalStorage;
 
   useEffect(() => {
-    setIsLocalStorageLoading(false);
-  }, [hasDetailsFromLocalStorage]);
+    const userEmail = user?.emailAddresses?.[0]?.emailAddress;
+
+    const updateUserDetailsFromHubspot = async (email: string) => {
+      const hubspotContact = await fetchHubspotContactDetails(email);
+      const subscriptionStatus = await getSubscriptionStatus(email);
+      setTermsInLocalStorage(true);
+      setValue("terms", true);
+      if (subscriptionStatus) {
+        setEmailInLocalStorage(email);
+        setValue("email", email);
+      } else {
+        setEmailInLocalStorage("");
+      }
+
+      if (hubspotContact) {
+        const schoolId = hubspotContact.schoolId;
+        const schoolName = hubspotContact.schoolName;
+
+        setSchoolInLocalStorage({
+          schoolId: schoolId ?? "",
+          schoolName: schoolName ?? "",
+        });
+
+        if (schoolName) {
+          setValue("schoolName", schoolName);
+        }
+
+        if (schoolId) {
+          setValue("school", schoolId);
+        }
+      } else {
+        setSchoolInLocalStorage({
+          schoolId: "",
+          schoolName: "",
+        });
+      }
+    };
+
+    if (userEmail && authFlagEnabled && isSignedIn) {
+      updateUserDetailsFromHubspot(userEmail);
```

Comment:
🏄🏻 `useEffect` doing network stuff always worries me (especially when it has a large dependencies array. There is a strong chance that this is/could be firing more than once unexpectedly. Something to watch for and hard to prevent.

Reply:
It just runs once when the page loads

Reply:
👍🏻 just something to be aware of. Hopefully no one will change the dependencies in a way that causes them to change on each render.

Reply:
Is there another safer way of doing this check ?

Reply:
Not easily with our current setup. It's too convenient to lean on `useEffect` as it's there, simple to understand and doesn't require any extra overhead. It's just not easy to predict when the dependencies array will be changed by an unrelated change and introduce a subtle bug.

Lots of [state](https://redux.js.org/) management [libraries](https://github.com/pmndrs/zustand) out there that formalise managing actions and state to make things predictable.

Ultimately the solution is to remove the side-effect from the component and put it somewhere else where it will only fire when its dependencies have been satisfied.

Currently you're dependent on at least 3 pieces of state (feature flag, signed in, user object) which load at different times (causing this component to render probably 3 times). This sucks. 

I can immediately see a potential bug, where `userEmail` might be `null` in the case of a race between this component rendering and Clerk loading.

If our state was managed centrally (ala Redux) we could avoid rendering this component until everything we needed was available and the app had transitioned into a stable state. We could then lift all of that logic in that effect into an action that only triggers when the app first enters that stable state.

For now, the best we can do is introduce refs/more local state that prevent the effect from firing more than once — but that only serves to make things even more complicated.